File: Source/Fantasy.Base/Tracking/TrackExtension.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace Fantasy.Tracking
{
    public static class TrackExtension
    {
        public static T GetProperty<T>(this ITrackProvider provider, string name, T defaultValue)
        {
            object obj = provider[name];
            if (obj != null)
            {
                return (T)obj;
               
            }

            else
            {

                return defaultValue;

            }
        }


        public static T GetProperty<T>(this ITrackListener listener, string name, T defaultValue)
        {
            object obj = listener[name];
            if (obj != null)
            {
                return (T)obj;

            }

            else
            {

                return defaultValue;

            }
        }
    }
}

File: Fantasy.Base/src/fantasy/tracking/TrackExtension.java
package fantasy.tracking;
@SuppressWarnings("unchecked")
public final class TrackExtension
{
//C# TO JAVA CONVERTER TODO TASK: Extension methods are not available in Java:
//ORIGINAL LINE: public static T GetProperty<T>(this ITrackProvider provider, string name, T defaultValue)
	
	public static <T> T GetProperty(ITrackProvider provider, String name, T defaultValue)
	{
		Object obj = provider.getItem(name);
		if (obj != null)
		{
			return (T)obj;

		}

		else
		{

			return defaultValue;

		}
	}


//C# TO JAVA CONVERTER TODO TASK: Extension methods are not available in Java:
//ORIGINAL LINE: public static T GetProperty<T>(this ITrackListener listener, string name, T defaultValue)
	public static <T> T GetProperty(ITrackListener listener, String name, T defaultValue)
	{
		Object obj = listener.getItem(name);
		if (obj != null)
		{
			return (T)obj;

		}

		else
		{

			return defaultValue;

		}
	}
}